### PR TITLE
feat: make beacon API port configurable

### LINKED ISSUE #811: Allow configuring the port which serves the Beacon API
Context: https://github.com/lambdaclass/lambda_ethereum_consensus/issues/172#issuecomment-1961920310

### PR DESCRIPTION
**Pull request by avilagaston9**
**Motivation**

This PR allows configuring the port that serves the Beacon API

**Description**

This PR adds the ```--beacon-api-port <PORT>``` flag, which accepts the port serving the Beacon API, allowing multiple nodes to run on each port.

Closes #811

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -11,7 +11,9 @@ switches = [
   testnet_dir: :string,
   metrics: :boolean,
   metrics_port: :integer,
-  log_file: :string
+  log_file: :string,
+  beacon_api_port: :integer,
+  no_halt: :boolean
```
[the remark this answers is not in the document]
Comment (MegaRedHand) [reply config/runtime.exs:17]: Is this needed?

[the remark this answers is not in the document]
Comment (avilagaston9) [reply config/runtime.exs:17]: Without it, the OptionParser raise an error "--no-halt : Unknown option" when running **mix phx.server**. Maybe there is a better alternative.

[the remark this answers is not in the document]
Comment (MegaRedHand) [reply config/runtime.exs:17]: Strange. I checked on my machine and that doesn't happen 🤔

**Review by MegaRedHand — APPROVED** [review]
Great work! Congrats on your first contribution 🥳

**Comment by MegaRedHand** [comment]
@all-contributors, would you be so kind as to add @avilagaston9 for code? thx

**Comment by MegaRedHand** [comment]
@all-contributors please add @avilagaston9 for code

**Pull request merged**